src/main/java/land/oras/Registry.java

```diff
@@ -739,19 +753,170 @@ private void logResponse(OrasHttpClient.ResponseWrapper<?> response) {
     }
 
     /**
-     * Push a blob using input stream to avoid loading the whole blob in memory
+     * Push a blob using input stream in chunks to avoid loading the whole blob in memory
      * @param containerRef the container ref
      * @param input the input stream
      * @param size the size of the blob
      * @return The Layer containing the uploaded blob information
      * @throws OrasException if upload fails or digest calculation fails
      */
-    public Layer pushBlobStream(ContainerRef containerRef, InputStream input, long size) {
+    public Layer pushChunks(ContainerRef containerRef, InputStream input, long size) {
+        // We Initilase the Message Digest first
+        MessageDigest digest;
+        try {
+            digest = MessageDigest.getInstance(
```

Comment:
Logic should not be here

Use 

```java
public static boolean isSupported(String prefix)
```

Or implement the logic in private method